account_billing/models/account_billing.py

```diff
@@ -0,0 +1,202 @@
+            return {'domain': {'partner_id': [
+                (MAP_INVOICE_TYPE_PARTNER_TYPE[self.bill_type], '=', True)]}}
+
+    @api.multi
+    def _get_partner_id(self):
+        partner_ids = self.env['account.invoice'].browse(
+            self._context.get('active_ids', [])).mapped('partner_id')
+        if len(partner_ids) > 1:
+            raise ValidationError(
+                _('Please select invoices with same partner'))
+        return partner_ids
+
+    @api.multi
+    def _get_currency_id(self):
+        currency_ids = self.env['account.invoice'].browse(
+            self._context.get('active_ids', [])).mapped('currency_id')
+        if currency_ids:
+            if len(currency_ids) > 1:
+                raise ValidationError(
+                    _('Please select invoices with same currency'))
+            elif currency_ids != self.env.user.company_id.currency_id:
+                return currency_ids
+        return self.env.user.company_id.currency_id
+
+    @api.multi
+    def _compute_invoice_related_count(self):
+        for billing in self:
+            invoice_ids = self.env['account.invoice'].search([
+                ('id', 'in', billing.invoice_ids.ids)
+            ])
+            billing.invoice_related_count = len(invoice_ids)
+
+    @api.multi
+    def name_get(self):
+        result = []
+        for billing in self:
+            result.append((billing.id, (billing.name or 'Draft')))
+        return result
+
+    @api.multi
+    def validate_billing(self):
+        for rec in self:
+            if any(inv.state != 'open' for inv in rec.invoice_ids):
+                raise ValidationError(_('Billing cannot be processed because \
+                    some invoices are not in state Open'))
+            # keep the number in case of a billing reset to draft
+            if not rec.name:
+                # Use the right sequence to set the name
+                if rec.bill_type == 'out_invoice':
+                    sequence_code = 'account.customer.billing'
+                if rec.bill_type == 'in_invoice':
+                    sequence_code = 'account.supplier.billing'
+                rec.name = self.env['ir.sequence'].with_context(
+                    ir_sequence_date=rec.date).next_by_code(sequence_code)
+            rec.write({'state': 'billed'})
+            rec.message_post(body=_('Billing is billed.'))
+        return True
+
+    @api.multi
+    def action_cancel_draft(self):
+        for rec in self:
+            rec.write({'state': 'draft'})
+            rec.message_post(body=_('Billing is reset to draft'))
+        return True
+
+    @api.multi
+    def action_cancel(self):
+        for rec in self:
+            invoice_paid = rec.invoice_ids.filtered(
+                lambda l: l.state == 'paid')
+            if invoice_paid:
+                raise ValidationError(_("Invoice paid already."))
+            rec.write({'state': 'cancel'})
+            for invoice in rec.invoice_ids:
```

Comment:
Can we use
`rec.invoice_ids.write({'billing_id': False})`
Note: why do we use update() method anyway?